Crash: InflateException

### Checked for duplicates?

- [x] This issue is not a duplicate

### Does it also happen in the desktop version?

- [ ] This bug does not occur in the latest version of Anki Desktop

### What are the steps to reproduce this bug?

1.Open the app
2.Enter a deck
3.Open the whiteboard
4.Select the color picker option

https://github.com/user-attachments/assets/4f30f451-e757-4f09-898f-844b92eb6d1a

### Expected behaviour

the app crashed

### Debug info

```text
AnkiDroid Version = 2.24.0alpha10-debug (b46eacd5fae013d8901547162e3c45540f9da8c1)  
Backend Version = 0.1.63-anki25.09.2 (25.09.2 3890e12c9e48c028c3f12aa58cb64bd9f8895e30)  
Android Version = 9 (SDK 28)  
ProductFlavor = full  
Device Info = Google | google | generic_x86_arm | sdk_gphone_x86_arm | AOSP on IA Emulator | ranchu  
WebView Info = [com.android.chrome | 349710017]: Mozilla/5.0 (Linux; Android 9; AOSP on IA Emulator Build/PSR1.180720.117; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/69.0.3497.100 Mobile Safari/537.36  
ACRA UUID = 81bb560e-0aa4-4093-be78-fcd10db40072  
FSRS = 5.1.0 (Enabled: false)  
Crash Reports Enabled = false
```

### (Optional) Anything else you want to share?

the crash log is here
action: 2@back
action: 1@click content-desc=เปิดแถบเมนูข้าง :android.widget.ImageButton@
action: 2@click className=android.widget.CheckedTextView,instance=1 :android.widget.CheckedTextView@recycler items_com.ichi2.anki.DeckPicker_000000000
action: 12@back
action: 1@click content-desc=เปิดแถบเมนูข้าง :android.widget.ImageButton@
action: 2@click className=android.widget.CheckedTextView,instance=1 :android.widget.CheckedTextView@recycler items_com.ichi2.anki.DeckPicker_000000000
action: 4@click resource-id=com.ichi2.anki.debug:id/action_add_note_from_card_browser :android.widget.Button@
action: 2@click resource-id=com.ichi2.anki.debug:id/action_save :android.widget.Button@
action: 12@click resource-id=com.ichi2.anki.debug:id/toggle_sticky :android.widget.ImageButton@
action: 15@edit resource-id=com.ichi2.anki.debug:id/edit_text :android.widget.EditText@
action: 2@click resource-id=com.ichi2.anki.debug:id/action_save :android.widget.Button@
action: 10@click resource-id=com.ichi2.anki.debug:id/expand_button :android.widget.ImageButton@
action: 17@back
action: 11@click resource-id=com.ichi2.anki.debug:id/toggle_sticky :android.widget.ImageButton@
action: 2@click resource-id=com.ichi2.anki.debug:id/action_save :android.widget.Button@
action: 11@click resource-id=com.ichi2.anki.debug:id/toggle_sticky :android.widget.ImageButton@
action: 9@click resource-id=com.ichi2.anki.debug:id/media_button :android.widget.ImageButton@
action: 7@click className=android.widget.TextView,instance=2 :android.widget.TextView@આકૃતિ
action: 4@click resource-id=com.ichi2.anki.debug:id/redo_button :android.widget.ImageButton@
action: 4@click resource-id=com.ichi2.anki.debug:id/redo_button :android.widget.ImageButton@
action: 5@click resource-id=com.ichi2.anki.debug:id/overflow_menu_button :android.widget.ImageButton@
action: 1@click content-desc=กลับ :android.widget.ImageButton@
action: 10@click resource-id=com.ichi2.anki.debug:id/icon :android.widget.ImageView@
03-24 23:44:45.121 28981 28981 E AndroidRuntime: FATAL EXCEPTION: main
03-24 23:44:45.121 28981 28981 E AndroidRuntime: Process: com.ichi2.anki.debug, PID: 28981
03-24 23:44:45.121 28981 28981 E AndroidRuntime: android.view.InflateException: Binary XML file line #30: Binary XML file line #30: Error inflating class com.skydoves.colorpickerview.AlphaTileView
03-24 23:44:45.121 28981 28981 E AndroidRuntime: Caused by: android.view.InflateException: Binary XML file line #30: Error inflating class com.skydoves.colorpickerview.AlphaTileView
03-24 23:44:45.121 28981 28981 E AndroidRuntime: Caused by: java.lang.reflect.InvocationTargetException
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at java.lang.reflect.Constructor.newInstance0(Native Method)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.view.LayoutInflater.createView(LayoutInflater.java:647)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:790)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:730)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.view.LayoutInflater.rInflate(LayoutInflater.java:863)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.view.LayoutInflater.inflate(LayoutInflater.java:515)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.view.LayoutInflater.inflate(LayoutInflater.java:374)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at com.skydoves.colorpickerview.flag.FlagView.initializeLayout(FlagView.java:70)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at com.skydoves.colorpickerview.flag.FlagView.<init>(FlagView.java:39)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at com.ichi2.anki.ui.windows.reviewer.whiteboard.BubbleFlag.<init>(ColorPickerDialog.kt:92)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at com.ichi2.anki.ui.windows.reviewer.whiteboard.ColorPickerDialogKt.showColorPickerDialog(ColorPickerDialog.kt:62)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at com.ichi2.anki.ui.windows.reviewer.whiteboard.WhiteboardFragment.showAddColorDialog(WhiteboardFragment.kt:250)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at com.ichi2.anki.ui.windows.reviewer.whiteboard.WhiteboardFragment.onMenuItemClick(WhiteboardFragment.kt:417)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at androidx.appcompat.widget.PopupMenu$1.onMenuItemSelected(PopupMenu.java:113)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at androidx.appcompat.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:833)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at androidx.appcompat.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:157)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:984)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at androidx.appcompat.view.menu.MenuPopup.onItemClick(MenuPopup.java:128)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.widget.AdapterView.performItemClick(AdapterView.java:318)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.widget.AbsListView.performItemClick(AbsListView.java:1159)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.widget.AbsListView$PerformClick.run(AbsListView.java:3136)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.widget.AbsListView$3.run(AbsListView.java:4052)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:873)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:99)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:193)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6669)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: Caused by: android.content.res.Resources$NotFoundException: Drawable (missing name) with resource ID #0x7f0800c3
03-24 23:44:45.121 28981 28981 E AndroidRuntime: Caused by: android.content.res.Resources$NotFoundException: Unable to find resource ID #0x7f0800c3
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.content.res.ResourcesImpl.getResourceName(ResourcesImpl.java:255)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.content.res.ResourcesImpl.loadDrawableForCookie(ResourcesImpl.java:785)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.content.res.ResourcesImpl.loadDrawable(ResourcesImpl.java:631)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.content.res.Resources.loadDrawable(Resources.java:897)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.content.res.TypedArray.getDrawableForDensity(TypedArray.java:955)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.content.res.TypedArray.getDrawable(TypedArray.java:930)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.view.View.<init>(View.java:5010)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.view.View.<init>(View.java:4911)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.view.View.<init>(View.java:4890)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at com.skydoves.colorpickerview.AlphaTileView.<init>(AlphaTileView.java:46)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at java.lang.reflect.Constructor.newInstance0(Native Method)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.view.LayoutInflater.createView(LayoutInflater.java:647)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:790)
03-24 23:44:45.121 28981 28981 E AndroidRuntime: 	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:730)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at android.view.LayoutInflater.rInflate(LayoutInflater.java:863)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at android.view.LayoutInflater.inflate(LayoutInflater.java:515)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at android.view.LayoutInflater.inflate(LayoutInflater.java:374)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at com.skydoves.colorpickerview.flag.FlagView.initializeLayout(FlagView.java:70)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at com.skydoves.colorpickerview.flag.FlagView.<init>(FlagView.java:39)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at com.ichi2.anki.ui.windows.reviewer.whiteboard.BubbleFlag.<init>(ColorPickerDialog.kt:92)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at com.ichi2.anki.ui.windows.reviewer.whiteboard.ColorPickerDialogKt.showColorPickerDialog(ColorPickerDialog.kt:62)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at com.ichi2.anki.ui.windows.reviewer.whiteboard.WhiteboardFragment.showAddColorDialog(WhiteboardFragment.kt:250)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at com.ichi2.anki.ui.windows.reviewer.whiteboard.WhiteboardFragment.onMenuItemClick(WhiteboardFragment.kt:417)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at androidx.appcompat.widget.PopupMenu$1.onMenuItemSelected(PopupMenu.java:113)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at androidx.appcompat.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:833)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at androidx.appcompat.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:157)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:984)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at androidx.appcompat.view.menu.MenuPopup.onItemClick(MenuPopup.java:128)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at android.widget.AdapterView.performItemClick(AdapterView.java:318)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at android.widget.AbsListView.performItemClick(AbsListView.java:1159)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at android.widget.AbsListView$PerformClick.run(AbsListView.java:3136)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at android.widget.AbsListView$3.run(AbsListView.java:4052)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:873)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:99)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:193)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6669)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
03-24 23:44:45.122 28981 28981 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)

### Research

- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

--- a/AnkiDroid/src/main/java/com/ichi2/anki/ui/windows/reviewer/whiteboard/ColorPickerDialog.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/ui/windows/reviewer/whiteboard/ColorPickerDialog.kt
@@ -17,8 +17,12 @@
 package com.ichi2.anki.ui.windows.reviewer.whiteboard
 
 import android.content.Context
+import android.os.Build
 import android.view.Choreographer
+import android.widget.FrameLayout
 import androidx.annotation.ColorInt
+import androidx.core.view.updateLayoutParams
+import androidx.core.view.updatePadding
 import com.ichi2.anki.R
 import com.ichi2.utils.Dp
 import com.ichi2.utils.dp
@@ -59,7 +63,24 @@ fun Context.showColorPickerDialog(
             // default position, resulting in an incorrect displayed color.
             colorPickerView.post { colorPickerView.setInitialColor(initialColor) }
             // Bubble showing the selected color
-            colorPickerView.flagView = BubbleFlag(this@showColorPickerDialog)
+            val bubbleFlag = BubbleFlag(this@showColorPickerDialog)
+            colorPickerView.flagView = bubbleFlag
+            // TODO on these lower Android versions the view clipping fails and we end up with a
+            //  square(standard view shape) instead of the expected circle so we update the
+            //  positioning values so the square sis smaller and is centered in the background pin
+            if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.R) {
+                bubbleFlag
+                    .findViewById<FrameLayout>(R.id.flag_color_layout_wrapper)
+                    .updatePadding(
+                        10.dp.toPx(this@showColorPickerDialog),
+                        7.dp.toPx(this@showColorPickerDialog),
+                        10.dp.toPx(this@showColorPickerDialog),
+                    )
+                bubbleFlag.alphaTileView.updateLayoutParams {
+                    this.width = 12.dp.toPx(this@showColorPickerDialog)
+                    this.height = 12.dp.toPx(this@showColorPickerDialog)
+                }
+            }
             setTitle(R.string.choose_color)
             setPositiveButton(R.string.dialog_ok, onPositiveCallback)
             negativeButton(R.string.dialog_cancel)

--- a/AnkiDroid/src/main/res/drawable/color_picker_circle_indicator.xml
+++ b/AnkiDroid/src/main/res/drawable/color_picker_circle_indicator.xml
@@ -1,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="oval">
+    <!-- color doesn't really matter -->
+    <solid android:color="@color/material_blue_500" />
+</shape>

--- a/AnkiDroid/src/main/res/layout/colorpicker_flag_bubble.xml
+++ b/AnkiDroid/src/main/res/layout/colorpicker_flag_bubble.xml
@@ -15,17 +15,18 @@
   ~  this program.  If not, see <http://www.gnu.org/licenses/>.
   -->
 <FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:id="@+id/flag_color_layout_wrapper"
     android:layout_width="wrap_content"
     android:layout_height="wrap_content"
     android:paddingHorizontal="8dp"
     android:paddingTop="5dp"
     android:paddingBottom="11dp"
     android:background="@drawable/pin_shape">
-
+    <!-- TODO we already have a circle_background.xml but that one is used in widget theming? -->
     <com.skydoves.colorpickerview.AlphaTileView
         android:id="@+id/flag_color_layout"
         android:layout_width="16dp"
         android:layout_height="16dp"
-        android:background="@drawable/circle_background"
+        android:background="@drawable/color_picker_circle_indicator"
         android:clipToOutline="true" />
 </FrameLayout>

